docs/sdg/sdg-flow-yaml.md

```diff
@@ -0,0 +1,164 @@
+
+### CLI Integration
+
+As of the current version of `ilab`, it supports `simple` and `full` as parameters to `--pipeline` to select one of the two types of built-in pipelines included in the library. Once we have support for loading custom pipelines, we need a way for these to be specified with the CLI. Here is a proposed CLI UX for this:
+
+> `ilab data generate`
+
+Use the default pipeline, `simple`.
+
+> `ilab data generate --pipeline full`
+
+Use the built-in `full` pipeline.
+
+> `ilab data generate --pipeline full --pipeline-extend path/to/config.yaml`
+
+Use the `full` pipeline, but extend it with the pipeline defined in the provided config file.
+
+> `ilab data generate --pipeline none --pipeline-extend path/to/other_config.yaml`
+
+Do not use a built-in pipeline. Use only the pipeline in the custom configuration.
+
+### Future CLI Improvements
+
+A possible improvement would be to have a well-defined place on the filesystem where custom pipeline configs can be automatically loaded and included as options to the `--pipeline` parameter.
+
+For example, if the config format included new parameters, `name: full-extended` and `extends: full`, and the CLI discovered and loaded it automatically, we could support `--pipeline full-extended` without needing the additional `--pipeline-extend` option.
+
+`/usr/share/instructlab/sdg/` is a proposed location for this as a place for a distribution of InstructLab to include pre-defined custom pipelines, at least for Linux. See the [Filesystem Hierarchy Standard](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html) for more details on why this path is appropriate for this use-case.
+
+It would also make sense to support a configuration directory for user's own custom pipeline configurations. Assuming there is a base config directory, these could go in a `sdg` subdirectory. There is a separate proposal that discusses a proposed configuration location: <https://github.com/instructlab/dev-docs/pull/104>. Note this is separate from the distribution-provided, read-only pipelines discussed above with a different location.
+
+If we have a location with pipeline examples then a nice to have would be to have a `ilab data generate --list-pipelines`.
+
+## Alternative Approaches
+
+Alternatives already considered and discarded are listed below.
+
+### No Custom Flows
+
+It would be preferable to not support custom flows, especially so early in the project because:
+
+- We will need an extensive API to support this customization, and we will need to be careful about making incompatible changes to that API once it has been adopted.
+- We would learn more about the pipelines that users are creating if they were added to the library.
+
+This approach was discarded because of strong demand from downstream users to define custom flows to encapsulate proprietary pipeline configuration..
+
+### Custom Flows as Code
```

Comment:
I'd like to hear @russellb's comments on why this is no longer preferred as the time when https://github.com/instructlab/dev-docs/pull/98 was reviewed. in particular, what changed your mind to think YAML files are more preferable now?
I think some justification is helpful for the community to understand why opposite decision have been made during the development/design of the SDG library.

Reply:
I think I've got that [above](https://github.com/instructlab/dev-docs/pull/109#discussion_r1671708336) but of course happy to hear more from Russell

The change is whether the YAML format should be implemented on the CLI side or in the library, and I think Russell's initial preference for the former was because he was also thinking about the serving config ... something we haven't fully figured out yet, in fairness

In any case, I'll make an update to the doc to cover this

Reply:
I think this is resolved by commit 4425f8d